src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2405,9 +2485,33 @@ public void onChange(InetAddressAndPort endpoint, ApplicationState state, Versio
             {
                 switch (state)
                 {
+                    case RELEASE_VERSION:
+                        Nodes.nodes().update(endpoint, new CassandraVersion(value.value), NodeInfo::setReleaseVersion);
+                        break;
                     case DC:
+                        updateTopology(endpoint);
+                        Nodes.nodes().update(endpoint, value.value, NodeInfo::setDataCenter);
+                        break;
                     case RACK:
                         updateTopology(endpoint);
+                        Nodes.nodes().update(endpoint, value.value, NodeInfo::setRack);
+                        break;
+                    case RPC_ADDRESS:
+                    case NATIVE_ADDRESS_AND_PORT:
+                        try
+                        {
+                            Nodes.nodes().update(endpoint, InetAddressAndPort.getByName(value.value), NodeInfo::setNativeTransportAddressAndPort);
+                        }
+                        catch (UnknownHostException e)
+                        {
+                            throw new RuntimeException(e);
+                        }
+                        break;
+                    case SCHEMA:
+                        Nodes.nodes().update(endpoint, UUID.fromString(value.value), NodeInfo::setSchemaVersion);
+                        break;
+                    case HOST_ID:
+                        Nodes.nodes().update(endpoint, UUID.fromString(value.value), NodeInfo::setHostId);
```

Comment:
I think we can move the INDEX_STATUS clause here as well now, because we will update the index status on joining the ring, so no need to do that earlier.